app/controllers/api/v2/CorsProxyController.ts

```diff
@@ -0,0 +1,64 @@
+import { Readable } from 'stream';
+import { ReadableStream } from 'stream/web';
+
+import { Context } from 'koa';
+import { Duration } from 'luxon';
+
+import { BadRequestException, ValidationException } from '../../../support/exceptions';
+import { currentConfig } from '../../../support/app-async-context';
+
+const headersToPass = ['Location', 'Content-Type', 'Content-Length'];
+const fallbackTimeoutMs = 1000;
+
+export async function proxy(ctx: Context) {
+  const {
+    timeout: timeoutString,
+    allowedOrigins,
+    allowedURlPrefixes,
+    allowLocalhostOrigins,
+  } = currentConfig().corsProxy;
+
+  const { origin } = ctx.headers;
+
+  if (typeof origin !== 'string') {
+    if (!allowedOrigins.includes('none')) {
+      throw new BadRequestException('Missing origin');
+    }
+  } else if (
+    !(
+      allowedOrigins.includes(origin) ||
+      (allowLocalhostOrigins && /^https?:\/localhost(:\d+)?$/.test(origin))
+    )
+  ) {
+    throw new BadRequestException('Origin not allowed');
+  }
+
+  let { url } = ctx.request.query;
+
+  if (Array.isArray(url)) {
+    [url] = url;
+  }
+
+  if (typeof url !== 'string') {
+    throw new ValidationException("Missing 'url' parameter");
+  }
+
+  if (!allowedURlPrefixes.some((prefix) => url.startsWith(prefix))) {
+    throw new ValidationException('URL not allowed');
+  }
+
+  const timeoutDuration = Duration.fromISO(timeoutString);
+  const timeoutMs = timeoutDuration.isValid ? timeoutDuration.toMillis() : fallbackTimeoutMs;
+
+  const response = await fetch(url, { signal: AbortSignal.timeout(timeoutMs) });
+
+  ctx.status = response.status;
+
+  for (const header of headersToPass) {
```

Comment:
It's the only place where headersToPass is used. Maybe it's better to inline it here? Will be more readable. `[...].forEach((header) => {})`